### PR TITLE
Port ure logger interface to guile by defining scm functions.

### PR DESCRIPTION
**Pull request by shujingke**

### CODE REVIEW

**Comment by shujingke** [comment]
It seems there is not a list of documentation of ure scheme interfaces in wiki. Maybe we should add the ure section into this page: https://wiki.opencog.org/wikihome/index.php/Scheme

File: opencog/scm/opencog/rule-engine/rule-engine-utils.scm [hunk 1]
```diff
@@ -313,6 +314,34 @@
 "
   (map (lambda (x) (gen-rand-variable prefix base length)) (iota n)))
 
+(define (set-ure-logger-level! level)
```
Comment (ngeiswei) [inline opencog/scm/opencog/rule-engine/rule-engine-utils.scm]: I think function names like `ure-logger-set-level!` would be better as they would be more consistent with the existing function names. What do you think?

**Comment by AmeBel** [comment]
@ngeiswei what do you mean by "subroutings of the URE "

**Comment by ngeiswei** [comment]
@AmeBel I mean functions for
1. Unifying a rule with a conclusion or premise
2. Expanding an inference tree backward or forward
3. Selecting an inference tree from the inference tree pool
4. Selecting a premise from an inference tree
5. Selecting a valid rule to expand an inference tree

probably other functions as well. This way a user should be able to implement it's own URE in just a few lines of Scheme code.

**Comment by shujingke** [comment]
OK I changed the names. See the new pull request.

**Pull request closed without merging**